src/main/java/usecase/alter_entry/AlterEntryController.java

```diff
@@ -0,0 +1,15 @@
+package usecase.alter_entry;
+
+import billgates.database.QueryEntryData;
+
+public class AlterEntryController {
+    private final AlterEntryInputPort useCase;
+
+    public AlterEntryController(AlterEntryInputPort useCase) {
+        this.useCase = useCase;
+    }
+
+    public void update(int billId, QueryEntryData entry) {
```

Comment:
`QueryEntryData` should not be used here.

`QueryEntryData` is used as the return value of the database gateway. It is the consequence of a query. 

In this case, the controller accepts the user's inputs and there's no query happens.

Instead, you should pass the id, altered column, and new value of the altered entry here.

For example, I altered the date column of the id 1 entry. The use case should alter the date column of the entry with id = 1 in the current bill by the new date.

Reply:
already fixed, please kindly check it!